src/main/java/org/apache/sysml/runtime/controlprogram/paramserv/ParamServer.java

```diff
@@ -180,52 +197,65 @@ private void setFinishedState(int workerID) {
 			_finishedStates[workerID] = true;
 		}
 
+		private void broadcastModel() {
+			IntStream.range(0, _finishedStates.length).forEach(i -> _modelMap.compute(i, (id, q) -> {
+				if (q == null) {
+					q = new ArrayBlockingQueue<>(1);
+				}
+				try {
+					q.put(_model);
+				} catch (InterruptedException e) {
+					throw new DMLRuntimeException(
```

Comment:
I usually use streams only when the called methods throw runtime exceptions - for other exceptions a for loop without explicit exception handling is probably simpler.